chips/cc26x0/src/uart.rs

```diff
@@ -174,15 +243,74 @@ impl UART {
         regs.icr.write(Interrupts::ALL_INTERRUPTS::SET);
     }
 
+    pub fn enable_interrupts(&self) {
+        let regs = unsafe { &*self.regs };
+        // Clear all UART interrupts
+        regs.icr.write(Interrupts::ALL_INTERRUPTS::SET);
+        // Enable all UART interrupts        
+        regs.imsc.write(Interrupts::CTSM::SET + 
+                        Interrupts::RX::SET   + 
+                        Interrupts::TX::SET   + 
+                        Interrupts::RT::SET   + 
+                        Interrupts::FE::SET   + 
+                        Interrupts::PE::SET   + 
+                        Interrupts::BE::SET   + 
+                        Interrupts::OE::SET);
+    }
+
+
     pub fn handle_interrupt(&self) {
         let regs = unsafe { &*self.regs };
         // Get status bits
         #[allow(unused)]
         let flags: u32 = regs.fr.get();
+
         // Clear interrupts
         regs.icr.write(Interrupts::ALL_INTERRUPTS::SET);
+        regs.rsr_ecr.write(Errors::ALL_ERRORS::SET);
+
+        self.disable_interrupts();
+
+        let tx_transfer_complete = unsafe{udma::UDMA.transfer_complete(TX_DMA)};
+        
+        if tx_transfer_complete {
+            regs.dmactl.modify(DMACtl::TXDMAE::CLEAR);
+            //clear the transfer flag
+            unsafe{udma::UDMA.clear_transfer(TX_DMA)};
+
+            self.client.get().map(|client| {
+                self.tx_buffer.take().map(|tx_buffer| {
+                    client.transmit_complete(
+                        tx_buffer,
+                        kernel::hil::uart::Error::CommandComplete,
+                    );
+                });
+            });
+        }
+
+        let rx_transfer_complete = unsafe{udma::UDMA.transfer_complete(RX_DMA)};
+
```

Comment:
Can `rx_transfer_complete` and `tx_transfer_complete` both be true?

Reply:
I think so- they're separate channels in the DMA controller so they can both be running, and while they point to the same register (UART DR), one is purely reading the other writing.